packages/wallet/test/e2e/SingleAddressWallet.test.ts

```diff
@@ -32,97 +29,154 @@ const createDelegationCertificates = async (wallet: Wallet, rewardAccount: Carda
+        expect(afterTx1TotalBalance.two.coins).toBe(expectedCoinsAfterTx1.two);
+        const utxo = walletOne.utxo.total$.value!;
+        const expectedCoinsWhileTxPending = {
+          one:
+            initialTotalBalance.one.coins -
+            Cardano.util.coalesceValueQuantities(
+              tx1Internals.body.inputs.map((txInput) => utxo.find(([txIn]) => txIn.txId === txInput.txId)![1].value)
+            ).coins,
+          two: initialTotalBalance.two.coins
+        };
+
+        expect(afterTx1AvailableBalance.one.coins).toBe(expectedCoinsWhileTxPending.one);
+        expect(afterTx1AvailableBalance.two.coins).toBe(expectedCoinsWhileTxPending.two);
+        await waitForBalanceCoins(expectedCoinsAfterTx1.one);
+        await waitForBalanceCoins(expectedCoinsAfterTx1.two);
       })(),
-      // Test it updates wallet.delegation after delegating to stake pool
+      // Test it updates walletOne.delegation after delegating to stake pool
       (async () => {
-        const delegateeAfter1stTx = await firstValueFrom(wallet.delegation.delegatee$.pipe(skip(1)));
+        const delegateeAfter1stTx = await firstValueFrom(walletOne.delegation.delegatee$.pipe(skip(1)));
         expect(delegateeAfter1stTx.nextNextEpoch?.id).toBe(poolId);
       })()
     ]);
 
-    // Make a 2nd tx with key deregistration
-    const tx2Internals = await wallet.initializeTx({
+    // Make a 2nd tx with:
+    // - walletOne key deregistration
+    const tx2Internals = await walletOne.initializeTx({
       certificates: [{ __typename: Cardano.CertificateType.StakeKeyDeregistration, address: rewardAccount }]
     });
-    await wallet.submitTx(await wallet.finalizeTx(tx2Internals));
+    await walletOne.submitTx(await walletOne.finalizeTx(tx2Internals));
 
     // No longer delegating
-    const delegateeAfter2ndTx = await firstValueFrom(wallet.delegation.delegatee$.pipe(skip(1)));
+    const delegateeAfter2ndTx = await firstValueFrom(walletOne.delegation.delegatee$.pipe(skip(1)));
     expect(delegateeAfter2ndTx.nextNextEpoch?.id).toBeUndefined();
 
-    // Deposit is returned to wallet balance
-    await waitForBalanceCoins(expectedCoinsAfterTx1 + stakeKeyDeposit - tx2Internals.body.fee);
-    expect((await firstValueFrom(wallet.balance.total$)).deposit).toBe(0n);
+    // Deposit is returned to walletOne balance
+    await waitForBalanceCoins(expectedCoinsAfterTx1.one + stakeKeyDeposit - tx2Internals.body.fee);
+    expect((await firstValueFrom(walletOne.balance.total$)).deposit).toBe(0n);
+
+    const afterTx2TotalBalance = {
+      one: await firstValueFrom(walletOne.balance.total$),
+      two: await firstValueFrom(walletTwo.balance.total$)
+    };
+
+    // Make a 3rd tx using wallet2 with:
+    // - payment to return the coins received in tx1 for future tests
+
+    const tx3Internals = await walletTwo.initializeTx({
+      outputs: new Set([{ address: walletOne.addresses[0].bech32, value: { coins: tx1OutputCoins } }])
```

Comment:
Need to send smaller quantity to still have funds for fee if walletTwo was originally empty, which is likely to be the case for new test setups (faucet limits to 1 request per 24h or sth).

Reply:
I've loaded both CI wallets with 1000 ada